Package-customised images and alpine-core variant images

piotrminkina opened the pull request:
I implemented support for build package-customised images, prepared alpine-core variant images (without init system, provided by `openrc` and `busybox-initscripts` packages) and updated edge to 3.2 RC2 version with all image variants.

andyshinn commented:
Interesting. The customized packages for builder is fine. But can you:

* Explain what core is and how it compares to the alpine-base package. Is this an official thing?
* Remove 3.2. We'll get that added once it is officially released (edge is same as 3.2 at the moment).
* Check the build failures and add in the necessary test files.

piotrminkina commented:
Hello @andyshinn 

Alpine-core is unofficial and it is only my own codename. Package alpine-base also installs `alpine-conf`, `busybox-initscripts` and `openrc` that are unnecessary and useless in containerized environments.

I removed 3.2 and fixed tests.

andyshinn commented:
Cool. I'll play with this next week.

piotrminkina commented:
I also observed that core images are up to 9% smaller from its original equivalents.

andyshinn commented:
I wonder if there is any reason to have two images. Do you think that "core" could just be the main image? Do you have any use case where you want both? I agree that a Docker container environment has no need for OpenRC, init scripts, and configuration (especially when it can be added manually afterwards).

I'm still looking for a good way to test a handful of images to make sure they work properly on "core" before making this decision. @progrium @mattaitchison thoughs?

piotrminkina commented:
I think use core image as main image is a good idea. I think also if people want `openrc`, then they will prefer `runkit`, `supervisord` or something like.

progrium commented:
@andyshinn you mean test some existing images built with alpine to see if using just core is enough to be backwards compatible? maybe make an `experimental` branch that can be used for tests like this. then we can switch some apps to it and see that they work just fine. then we can decide to do it in master.

piotrminkina commented:
Any plans to merge this PR?

piotrminkina commented:
Rebased  with the newest changes from master.

andyshinn commented:
Can you modify this to just be the default in the images (remove the core
variant altogether)? CircleCI will then generate some images I can test
with.

My plan is to run them through Deis testing and see if it passes. Deis is
running Alpine for their stuff now and should be a good benchmark.
On Jul 4, 2015 5:00 AM, "Piotr Minkina" <notifications@github.com> wrote:

> Rebased with the newest changes from master.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gliderlabs/docker-alpine/pull/32#issuecomment-118496866>
> .
>

piotrminkina commented:
Yes, of course :) done.

envygeeks commented:
Is there any status on getting this into a branch?

envygeeks commented:
I just wanted to point out that this update excludes important pieces like /etc/{os-release,issue} so if you plan to integrate this you should probably download the base manually and unpack those files into the image without installing the entire package.

piotrminkina commented:
It's doint at this place: https://github.com/gliderlabs/docker-alpine/pull/32/files#diff-c0fb86435ae54de56f7dbb9f0eebb7d3R70

envygeeks commented:
@piotrminkina didn't even notice that part, any reason not to use the static binary to fetch it?

andyshinn commented:
@piotrminkina can we change the packages for the current versions to be the existing `alpine-base`? Then we can merge this in and plan on releasing the slimmer version with Alpine 3.3 or another pull request when we have some more time to test. I like the changes to the builder at least and would love to get this merged in.

piotrminkina commented:
Sure, but little later. I have some unfinished business.

andyshinn commented:
This is now superseded by #76 (which is merged in). I think we will introduce the "core" packages layout at Alpine 3.3 image time. Please feel free to re-open and rebase if we've missed something here other than the `-p` support in the builder.
